### PR TITLE
Finish content negotiation tests and start request body tests

### PR DESCRIPTION
**Pull request by ThatJoeMoore**

### CODE REVIEW

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt [hunk 1]
```diff
@@ -2,10 +2,36 @@ package edu.byu.uapi.server.http.integrationtest
...
 import edu.byu.uapi.server.http.integrationtest.dsl.ComplianceSpecSuite
 import edu.byu.uapi.server.http.integrationtest.dsl.SuiteDsl
+import edu.byu.uapi.server.http.integrationtest.dsl.TestResponse
+import edu.byu.uapi.server.http.integrationtest.dsl.expectTextBodyEquals
+import edu.byu.uapi.server.http.integrationtest.dsl.post
 
-object RequestBodySpecs: ComplianceSpecSuite() {
+/**
+ * The HTTP implementations should provide the handler with a representation of a body. The primary
+ * way for the handlers to interact with the body is using HttpRequestBody#
```
Comment (ThatJoeMoore) [inline views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: ..perhaps I should finish writing this comment....

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/ResponseHeaderSpecs.kt [hunk 2]
```diff
@@ -2,9 +2,26 @@ package edu.byu.uapi.server.http.integrationtest
...
+                ))
+            }
+        }
+        it("Should return the exact headers and values specified") {
+            whenCalledWith { get("") }
+            then {
+                expectHeaderWithValue("foo", "bar")
+                expectHeaderWithValue("BAZ", "zab")
+            }
+        }
```
Comment (ThatJoeMoore) [inline views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/ResponseHeaderSpecs.kt]: Tim Allen in Galaxy Quest: I don't like it. It was too easy

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt [hunk 3]
```diff
@@ -2,10 +2,36 @@ package edu.byu.uapi.server.http.integrationtest
...
+                    }
+                    TestResponse.Text("$invoked - $body")
+                }
+            }
+            whenCalledWith { post("") }
+            then {
+                expectTextBodyEquals("false - null")
+            }
+        }
+        describe("real bodies") {
```
Comment (jvisker) [inline views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: Did you mean to put something here?

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt (hunk 1 above)
Comment (jvisker) [reply views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: Do it?

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt [hunk 4]
```diff
@@ -2,10 +2,36 @@ package edu.byu.uapi.server.http.integrationtest
...
+ * The HTTP implementations should provide the handler with a representation of a body. The primary
+ * way for the handlers to interact with the body is using HttpRequestBody#
+ */
+object RequestBodySpecs : ComplianceSpecSuite() {
     override fun SuiteDsl.define() {
-        disabled = true
+        it("handles empty bodies") {
+            givenRoutes {
+                post {
+                    //consumeBody doesn't invoke the
```
Comment (yoshutch) [inline views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: doesn't invoke the post request?

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/dsl/RoutingMethods.kt [hunk 5]
```diff
@@ -2,22 +2,22 @@ package edu.byu.uapi.server.http.integrationtest.dsl
 
 import edu.byu.uapi.server.http.HttpMethod
 
-fun RoutingInit.get(consumes: String? = null, produces: String? = null, handler: TestHttpHandler) {
-    route(HttpMethod.Routable.GET, consumes, produces, handler)
+fun RoutingInit.get(consumes: String? = null, produces: String? = null, trackActualRequest: Boolean = true, handler: TestHttpHandler) {
```
Comment (yoshutch) [inline views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/dsl/RoutingMethods.kt]: Could you use the `DEFAULT_TRACK_ACTUAL_REQUEST ` const here or does it not make sense to?

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt (hunk 3 above)
Comment (ThatJoeMoore) [reply views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: Nope, it's not implemented yet (and shows up as a disabled test). It was trying to write those tests that got me going down my current beautiful rabbit hole.

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt (hunk 4 above)
Comment (ThatJoeMoore) [reply views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/RequestBodySpecs.kt]: Finished the comment.

File: views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/dsl/RoutingMethods.kt (hunk 5 above)
Comment (ThatJoeMoore) [reply views/http/common-test/src/main/kotlin/edu/byu/uapi/server/http/integrationtest/dsl/RoutingMethods.kt]: Done.

**Comment by ThatJoeMoore** [comment]
Closing in favor of #127 , as that obsoletes a lot of these changes.

**Pull request closed without merging**